### PR TITLE
#1425 #1422 #1441 Link to repo in package details should be clickable

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#1422 Package Manager UI: Library in details page should be clickable
#1441 Packaga manager: Move list of authors to the bottom
#1425 Link to repo in package details should be clickable

### CODE REVIEW

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs [hunk 1]
```diff
@@ -23,6 +23,9 @@ internal class RPackageViewModel : BindableBase, IRPackageViewModel {
         private bool _isUpdateAvailable;
 
         public static RPackageViewModel CreateAvailable(RPackage package) {
+            Uri uri = null;
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs]: Let's call it repositoryUri

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs [hunk 2]
```diff
@@ -55,6 +63,8 @@ internal class RPackageViewModel : BindableBase, IRPackageViewModel {
                 IsInstalled = true,
                 HasDetails = true
             };
+
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs]: Formatting?

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs [hunk 3]
```diff
@@ -30,11 +33,15 @@ internal class RPackageViewModel : BindableBase, IRPackageViewModel {
...
                 NeedsCompilation = package.NeedsCompilation != null && !package.NeedsCompilation.EqualsIgnoreCase("no"),
-                Repository = package.Repository,
+                RepositoryUri = uri,
+                RepositoryText = uri != null ? null : package.Repository,
             };
         }
 
         public static RPackageViewModel CreateInstalled(RPackage package) {
+            Uri uri = null;
+            Uri.TryCreate(package.Repository, UriKind.Absolute, out uri);
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs]: What will happen if `package.Repository` is null?

File: src/R/Components/Impl/PackageManager/Implementation/View/PackageDetails.xaml.cs [hunk 4]
```diff
@@ -14,15 +17,23 @@ public partial class PackageDetails : UserControl {
...
-            throw new System.NotImplementedException();
+            throw new NotImplementedException();
+        }
+
+        private void RepositoryUri_RequestNavigate(object sender, RequestNavigateEventArgs e) {
+            Process.Start(e.Uri.AbsoluteUri);
+        }
+
+        private void LibraryPath_RequestNavigate(object sender, RequestNavigateEventArgs e) {
+            Process.Start(e.Uri.ToString().Replace('/', '\\'));
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageDetails.xaml.cs]: @huguesv, is it guaranteed that `Package.LibPath` is URL?

Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageDetails.xaml.cs]: It's a path on disk in the following format:
```
> i1[1,'LibPath']
[1] "C:/Users/huvalo/Documents/R/win-library/3.2"
> 
```

Also, I've never seen it return a path with a ` in it, it's always expanded.  So the code above should work AFAIK.

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs (hunk 3 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs]: You get null URI so text will be used which is also null. You get Repositiry: and then empty

File: src/R/Components/Impl/PackageManager/Implementation/View/PackageDetails.xaml.cs (hunk 4 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageDetails.xaml.cs]: In the worst case it won't open anything in explorer

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs]: Fixed

**Comment by AlexanderSher** [comment]
:+1:

**Pull request merged**